spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -110,3111 +65,1034 @@
-	 * @param wireTapConfigurer the {@link Consumer} to accept options for the {@link WireTap}.
+	 * Use {@link #route(Class, Function)} if you need to access the entire message.
+	 * @param router the {@link Function} to use.
+	 * @param <S> the source payload type.
+	 * @param <T> the target result type.
 	 * @return the current {@link IntegrationFlowDefinition}.
 	 */
-	public B wireTap(MessageChannel wireTapChannel, Consumer<WireTapSpec> wireTapConfigurer) {
-		WireTapSpec wireTapSpec = new WireTapSpec(wireTapChannel);
-		if (wireTapConfigurer != null) {
-			wireTapConfigurer.accept(wireTapSpec);
-		}
-		addComponent(wireTapChannel);
-		return wireTap(wireTapSpec);
+	public <S, T> B route(Function<S, T> router) {
+		return route(null, router);
 	}
 
 	/**
-	 * Populate the {@code Wire Tap} EI Pattern specific
-	 * {@link org.springframework.messaging.support.ChannelInterceptor} implementation
-	 * to the current {@link #currentMessageChannel}.
-	 * <p> It is useful when an implicit {@link MessageChannel} is used between endpoints:
+	 * Populate the {@link MethodInvokingRouter} for provided {@link Function}
+	 * with provided options from {@link RouterSpec}.
+	 * In addition accept options for the integration endpoint using {@link GenericEndpointSpec}.
+	 * Typically used with a Java 8 Lambda expression:
 	 * <pre class="code">
 	 * {@code
-	 *  .transform("payload")
-	 *  .wireTap(new WireTap(tapChannel().selector(m -> m.getPayload().equals("foo")))
-	 *  .channel("foo")
+	 *  .<Integer, Boolean>route(p -> p % 2 == 0,
+	 *                 m -> m.channelMapping("true", "evenChannel")
+	 *                       .subFlowMapping("false", f ->
+	 *                                   f.<Integer>handle((p, h) -> p * 3))
+	 *                       .applySequence(false))
 	 * }
 	 * </pre>
-	 * This method can be used after any {@link #channel} for explicit {@link MessageChannel},
-	 * but with the caution do not impact existing {@link org.springframework.messaging.support.ChannelInterceptor}s.
-	 * @param wireTapSpec the {@link WireTapSpec} to use.
-	 * <p> When this EIP-method is used in the end of flow, it appends {@code nullChannel} to terminate flow properly,
-	 * Otherwise {@code Dispatcher has no subscribers} exception is thrown for implicit {@link DirectChannel}.
+	 * Use {@link #route(Class, Function, Consumer)} if you need to access the entire message.
+	 * @param router the {@link Function} to use.
+	 * @param routerConfigurer the {@link Consumer} to provide {@link MethodInvokingRouter} options.
+	 * @param <S> the source payload type.
+	 * @param <T> the target result type.
 	 * @return the current {@link IntegrationFlowDefinition}.
 	 */
+	public <S, T> B route(Function<S, T> router, Consumer<RouterSpec<T, MethodInvokingRouter>> routerConfigurer) {
+		return route(null, router, routerConfigurer);
+	}
+
+
+	// All the methods below override super for bytecode backward compatibility.
```

Comment:
I think we will need to mark all these with NOSONAR to avoid "useless override" smells from Sonar 😦

Reply:
Correct, but I would like to see that message to be sure about the place for such a `// NOSONAR`  comment